overcooked_ai_py/mdp/overcooked_env.py

```diff
@@ -18,11 +18,14 @@ class OvercookedEnv(object):
     it as the agent takes actions, and provides rewards to the agent.
     """
 
-    def __init__(self, mdp, start_state_fn=None, horizon=MAX_HORIZON, debug=False):
+    def __init__(self, mdp, start_state_fn=None, horizon=MAX_HORIZON, debug=False, hh_data_starts=False):
         """
         mdp (OvercookedGridworld or function): either an instance of the MDP or a function that returns MDP instances 
         start_state_fn (OvercookedState): function that returns start state for the MDP, called at each environment reset
         horizon (float): number of steps before the environment returns done=True
+
+        TODO: make the start state completely decided in the MDP. MDPs with different start states should be considered
```

Comment:
What would be the reason for doing this? Making a new MDP can take a while for some layouts, so on the rare occasion that lots of different start states need to be used (such as when doing the qualitative tests!) then having to make a new MDP each time might slow things down.

Reply:
I thought it would be a good idea because it makes it more compatible with AI Theory: an MDP is defined as having, among other things, a start state distribution. So just to make it more intuitive to people that know the theory, it might be nicer to keep the abstractions consistent with the theory.

Code wise, it also seems that dealing with start states in the mdp might make things a little bit more simple (but it does admittedly increase the burden on the MDP class, that is already quite big)

Reply:
Actually to be fair, if one wants to be pedantic about conforming to theory, we should also separate VariableMDPs into their own MDP class (and not have that handled by the OvercookedEnv), as VariableMDPs are themselves a specific type of MDP in which there is a distribution over decision process structure.

This might make sense to do eventually, as the variable MDP distinction is a bit annoying to deal with within the Env code, and introduces some issues around the place. But in light of this, it also seems plausible that both this and the start state changes would complicate things more than they simplify them.

Reply:
Ah okay, this makes sense!